clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/provider/agent/GoogleHealthCheckCachingAgent.groovy

```diff
@@ -0,0 +1,146 @@
+import com.netflix.spinnaker.clouddriver.google.cache.CacheResultBuilder
+import com.netflix.spinnaker.clouddriver.google.cache.Keys
+import com.netflix.spinnaker.clouddriver.google.model.GoogleHealthCheck
+import com.netflix.spinnaker.clouddriver.google.security.GoogleNamedAccountCredentials
+import groovy.util.logging.Slf4j
+
+import static com.netflix.spinnaker.cats.agent.AgentDataType.Authority.AUTHORITATIVE
+import static com.netflix.spinnaker.clouddriver.google.cache.Keys.Namespace.HEALTH_CHECKS
+
+@Slf4j
+class GoogleHealthCheckCachingAgent extends AbstractGoogleCachingAgent {
+
+  final Set<AgentDataType> providedDataTypes = [
+      AUTHORITATIVE.forType(HEALTH_CHECKS.ns)
+  ] as Set
+
+  String agentType = "$accountName/$GoogleHealthCheckCachingAgent.simpleName"
+
+  GoogleHealthCheckCachingAgent(String googleApplicationName,
+                                GoogleNamedAccountCredentials credentials,
+                                ObjectMapper objectMapper) {
+    super(googleApplicationName,
+          credentials,
+          objectMapper)
+  }
+
+  @Override
+  CacheResult loadData(ProviderCache providerCache) {
+    List<GoogleHealthCheck> httpHealthCheckList = loadHealthChecks()
+    buildCacheResult(providerCache, httpHealthCheckList)
+  }
+
+  List<GoogleHealthCheck> loadHealthChecks() {
+    List<GoogleHealthCheck> ret = []
+    def httpHealthChecks = compute.httpHealthChecks().list(project).execute().items as List
+    httpHealthChecks.each { HttpHealthCheck hc ->
+      ret << new GoogleHealthCheck(
+        name: hc.getName(),
+        healthCheckType: GoogleHealthCheck.HealthCheckType.HTTP,
+        port: hc.getPort(),
+        requestPath: hc.getRequestPath(),
+        checkIntervalSec: hc.getCheckIntervalSec(),
+        timeoutSec: hc.getTimeoutSec(),
+        healthyThreshold: hc.getHealthyThreshold(),
+        unhealthyThreshold: hc.getUnhealthyThreshold()
+      )
+    }
+
+    def httpsHealthChecks = compute.httpsHealthChecks().list(project).execute().items as List
+    httpsHealthChecks.each { HttpsHealthCheck hc ->
+      ret << new GoogleHealthCheck(
+        name: hc.getName(),
+        healthCheckType: GoogleHealthCheck.HealthCheckType.HTTPS,
+        port: hc.getPort(),
+        requestPath: hc.getRequestPath(),
+        checkIntervalSec: hc.getCheckIntervalSec(),
+        timeoutSec: hc.getTimeoutSec(),
+        healthyThreshold: hc.getHealthyThreshold(),
+        unhealthyThreshold: hc.getUnhealthyThreshold()
+      )
+    }
+
+    def healthChecks = compute.healthChecks().list(project).execute().items as List
+    healthChecks.each { HealthCheck hc ->
+      def newHC = new GoogleHealthCheck(
+        name: hc.getName(),
+        checkIntervalSec: hc.getCheckIntervalSec(),
+        timeoutSec: hc.getTimeoutSec(),
+        healthyThreshold: hc.getHealthyThreshold(),
+        unhealthyThreshold: hc.getUnhealthyThreshold()
+      )
+
+      switch(hc.getType()) {
+        case 'HTTP':
```

Comment:
Please add some comments in here about why you have to query for http health checks both here and above.